src/utils/stream_decrypt.ts

```diff
@@ -0,0 +1,39 @@
+import { createDecipheriv } from 'crypto';
+import { Transform } from 'stream';
+import { CipherIV, FileKey } from '../types';
+
+const algo = 'aes-256-gcm'; // crypto library does not accept this in uppercase. So gotta keep using aes-256-gcm
+const authTagLength = 16;
+
+export class StreamDecrypt extends Transform {
+	private encryptedData: Buffer = Buffer.from(''); // TODO: is it possible to pop the stream data in a buffer-like manner?
+
+	constructor(private readonly cipherIV: CipherIV, private readonly fileKey: FileKey) {
+		super();
+	}
+
+	_transform(chunk: Buffer, encoding: string, next: (err?: Error, data?: Buffer) => void): void {
+		if (encoding !== 'buffer') {
+			throw new Error('The encoding is not a buffer!');
+		}
+		this.encryptedData = Buffer.concat([this.encryptedData, chunk]);
```

Comment:
The standard accepts the chunk to be a `string` also; `encoding` has its own type
```suggestion
	_transform(chunk: Buffer | string, encoding: BufferEncoding, next: (err?: Error, data?: Buffer) => void): void {
		if (!(chunk instanceof Buffer)) {
			chunk = Buffer.from(chunk, encoding);
		}
		this.encryptedData = Buffer.concat([this.encryptedData, chunk]);
```